opentracing/open_tracer.go

```diff
@@ -0,0 +1,37 @@
+package opentracing
+
+// A simple, thin interface for Span creation. Though other implementations are
+// possible and plausible, most users will be fine with `NewStandardTracer()`.
+type OpenTracer interface {
+	TraceContextIDSource
+
+	// Create, start, and return a new Span with the given `operationName`, all
+	// without specifying a parent Span that can be used to incorporate the
+	// newly-returned Span into an existing trace.
+	//
+	// `keyValueTags` is a [possibly empty] list of <string, interface{}> tag
+	// pairs which are applied to the returned `Span`, just as if they were
+	// applied in successive calls to `Span.SetTag()`. If the even-numbered
+	// `keyValueTags` parameters are not of `string` type, the implementation
+	// is undefined (and may `panic()`).
+	//
+	// Examples:
+	//
+	//     val tracer OpenTracer = ...
+	//
+	//     sp := tracer.StartTrace("GetFeed")
+	//
+	//     sp := tracer.StartTrace(
+	//         "HandleHTTPRequest",
+	//         "user_agent", req.UserAgent,
+	//         "lucky_number", 42)
+	//
+	StartTrace(operationName string, keyValueTags ...interface{}) Span
```

Comment:
new name and signature here... note the `keyValueTags` idiom (which I've seen in a few places)... basically saves the programmer the trouble of typing something like `Tags{ ... }` and also degrades elegantly for the case where there are no tags to add.